test/optimizer/default_cost_model_test.cpp

```diff
@@ -0,0 +1,422 @@
+#include <utility>
+
+#include "gtest/gtest.h"
+#include "optimizer/cost_model/default_cost_model.h"
+#include "optimizer/operator_node.h"
+#include "optimizer/optimizer_context.h"
+#include "optimizer/physical_operators.h"
+#include "parser/expression/abstract_expression.h"
+#include "parser/expression/constant_value_expression.h"
+#include "type/transient_value.h"
+#include "type/transient_value_factory.h"
+
+#include "test_util/test_harness.h"
+
+namespace terrier::optimizer {
+class DefaultCostModelTests : public TerrierTest {
```

Comment:
This is an impressive amount of code. Can you please add some comments in it to explain how it works and what each test is doing? This is will make it easier for new people to maintain/extend after you graduate.